### ISSUE TITLE
Memory optimization

### ISSUE THREAD

**Issue by magicprinc** [eclipse-vertx/vert.x#4971]
There is a memory optimization that can be used in Vert.x in several places.
For example, here: 
`io.vertx.core.impl.VertxImpl.InternalTimerHandler`
`private final AtomicBoolean disposed = new AtomicBoolean();`
`AtomicBoolean` is quite a heavy object (many times bigger than volatile boolean)
It can be replaced with `VarHandle`. The `VarHandle` has all the same operations, but it is a static field member of a class and in object there is only a volatile byte or int field. You can see this technic in JDK 
`java.util.concurrent.FutureTask#set`

Replacing `Atomic`* with `VarHandle` and `volatile` in libraries makes sense.
It dramatically reduces the memory footprint without disadvantages.

I could make a PR for InternalTimerHandler.

**Comment by tsegismont** [comment]
yes perhaps that could be useful for `InternalTimerHandler` because we create a lot of them

@franz1981 are you aware of any performance implications on using Varhandle vs atomicboolean for CAS?

**Comment by magicprinc** [comment]
@tsegismont I am not an expert as famous @franz1981 is, but I can add: Atomic* uses it internally. You can copy working code directly from Atomic implementation. The only disadvantage I know: more code on your side

**Comment by magicprinc** [comment]
@franz1981 Could you help please

**Comment by franz1981** [comment]
I think https://github.com/openjdk/jdk/blob/701bc3bbbe49a46aea7efc195463cc2efd64a785/src/java.base/share/classes/jdk/internal/vm/SharedThreadContainer.java#L46 shows that you could use byte-level var handle, but if you use Java Object Layout to print the layout of AtomicBoolean, having less than int doesn't improve the footprint of the class (which are usually aligned at 8-16 bytes in size AND have different aligned layout based on the class header size). So they didn't see any value to change it to use a boolean/byte.

**Comment by re-thc** [comment]
I gave this a try if anyone's still interested.

**Comment by magicprinc** [comment]
FYI I have tried VarHandle + byte and boolean
Both work fine

https://github.com/softwaremill/jox/blob/main/structured/src/main/java/com/softwaremill/jox/structured/CancellableFork.java#L30
